hack/bundle-gen.py

```diff
@@ -0,0 +1,442 @@
+    parser.add_argument('--hold',
+                        default=False,
+                        help='Adds a /hold comment in commit body to prevent the PR from merging (use "/hold cancel" to remove)',
+                        action='store_true')
+    args = parser.parse_args()
+
+    if args.branch and args.branch == BRANCH_DEFAULT:
+        parser.error('--branch=master is assumed by this script and corresponds to the alpha channel')
+
+    if not os.path.isfile(args.registry_auth_file):
+        parser.error('--registry-auth-file ({}) does not exist, provide --registry-auth-file'.format(args.registry_auth_file))
+
+    if args.verbose:
+        global SUBPROCESS_REDIRECT
+        SUBPROCESS_REDIRECT = None
+
+    return args
+
+# build_and_push_image uses buildah to build the HIVE image (tagged with image_tag) and then pushes the image to quay
+def build_and_push_image(repo_dir, registry_auth_file, image_tag, branch, dry_run):
+    repo = git.Repo(repo_dir)
+    # sync up git repo with remote
+    try:
+        repo.git.checkout(branch)
+    except:
+        print('Failed to checkout branch {}'.format(branch))
+        raise
+
+    try:
+        repo.remotes.upstream.fetch()
+    except:
+        print('Failed to fetch upstream')
+        raise
+
+    # build/push the thing
+    container_name = '{}:{}'.format(OPERATORHUB_HIVE_IMAGE_DEFAULT, image_tag)
+    print('Building container {}'.format(container_name))
+    cmd = 'buildah bud --tag {} -f ./Dockerfile'.format(container_name).split()
+    subprocess.run(cmd)
+
+    if not dry_run:
+        print("Pushing container")
+        cmd = 'buildah push '
+        if registry_auth_file != None:
+            cmd = cmd + ' --authfile={} '.format(registry_auth_file)
+        cmd = cmd + ' {}'.format(container_name)
+        subprocess.run(cmd.split())
+
+# gen_hive_version generates and returns the hive version eg. "v1.2.3187-18827f6"
+def gen_hive_version(repo_dir, branch, commit_hash):
+    repo = git.Repo(repo_dir)
+
+    try:
+        repo.git.checkout(branch)
+    except:
+        print('Failed to checkout branch {}'.format(branch))
+        raise
+    # sha is the first 7 characters of commit_hash
+    sha = str(repo.rev_parse(commit_hash))[0:7]
+
+    # num commits is the number of git commits counted from the first commit to the provided commit_hash
+    # this is the equivalent of running:
+    # git rev-list `git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$"`..HEAD --count
+    parent = repo.git.rev_list('--max-parents=0', commit_hash)
+    num_commits = repo.git.rev_list('--count', '{parent}..{commit_hash}'.format(parent=parent, commit_hash=commit_hash))
+
+    return '{prefix}.{commits}-{sha}'.format(prefix=HIVE_VERSION_PREFIX, commits=num_commits, sha=sha)
+
+# bump_version_zstream bumps the zstream on provided prev_version
+# eg. bump_version_zstream('hive-operator.v1.0.11') = 'hive-operator.v1.0.12'
+def bump_version_zstream(prev_version):
```

Comment:
This is used to generate the version for a `--branch` whose commit doesn't match master. This covers branches <= ocm-2.3. When ocm-2.4 no longer tracks master for an update we would want to use `gen_hive_version` to bump the commit count.

Reply:
Need to special case that somehow where this is called.

Reply:
Committed the prefix swap separately.